### DISCUSSION TITLE
18650 cell

### DISCUSSION THREAD

**Discussion by akariwrynn** [pybamm-team/PyBaMM#2801] [Q&A]
Hi, I am trying to convert Chen2020 to a 18650 cell by changing the width and height of the electrode. However, when I apply a sinusoidal current, the simulation deletes some points and both current and voltage are not sinusoidal. When I change dt_max, it gives setup failed, calculating jacobian failed, and at the end gives try solving the model up to this time only or reducing dt_max error.  I also want to make an coin cell for my experiments but don't know how to do it.

Thanks

**Comment by brosaplanella** [comment]
That might have to do with the time stepping in the solution, can you share a minimum working example of your code?

**Reply by akariwrynn** [reply]
I will share it in a moment but I found out that it is working after -5C. At cold, the voltage and current becomes non-sinosodial but rather linear. Do you have any idea about this?

**Reply by akariwrynn** [reply]
params['Current function [A]']=current
    params['Ambient temperature [K]']=temperature+273.15
    params["Positive particle radius [m]"]=parameters.get("Positive particle radius")
    params["Electrode height [m]"]=0.065
    params["Electrode width [m]"]=0.018
    solver=pybamm.CasadiSolver("safe")
    sim = pybamm.Simulation(model, parameter_values=params,solver=solver)
    sim.solve(t_eval=time_array)

 current=lambda t: amplitude * pybamm.sin(2 * np.pi * frequency * t)
    time_array= np.linspace(0, period/frequency, points_per_period*period)
    t_dom_v,t_dom_c=simulation(current, time_array,parameters=parameters,mychemistry=chemistry,temperature=temperature)

Where simulation function contains all the parameters and run the simulation.

As I mentioned above, it works for temperatures above -5C but after that it cuts some points and the output voltage and current are not sin.

Thanks

**Reply by brosaplanella** [reply]
The code you are sharing does not work in my computer as the `simulation` function and the parameters are not defined. Can you please share a full example so I can test this on my end?
